python/tests/blocks.py

```diff
@@ -194,19 +196,35 @@ def test_copy(self):
             ],
             properties=Labels(["properties"], np.array([[5], [3]], dtype=np.int32)),
         )
-        copy = block.copy()
+
+        # using TensorBlock.copy
+        clone = block.copy()
         block_values_id = id(block.values)
 
         del block
 
-        assert id(copy.values) != block_values_id
+        assert id(clone.values) != block_values_id
+
+        assert_equal(clone.values, np.full((3, 3, 2), 2.0))
+        assert clone.samples.names == ("samples",)
+        assert len(clone.samples) == 3
+        assert tuple(clone.samples[0]) == (0,)
+        assert tuple(clone.samples[1]) == (2,)
+        assert tuple(clone.samples[2]) == (4,)
+
+        # using copy.deepcopy
+        other_clone = clone.copy()
+        block_values_id = id(clone.values)
+
+        del clone
+
+        assert id(other_clone.values) != block_values_id
+        assert_equal(other_clone.values, np.full((3, 3, 2), 2.0))
 
-        assert_equal(copy.values, np.full((3, 3, 2), 2.0))
-        assert copy.samples.names == ("samples",)
-        assert len(copy.samples) == 3
-        assert tuple(copy.samples[0]) == (0,)
-        assert tuple(copy.samples[1]) == (2,)
-        assert tuple(copy.samples[2]) == (4,)
+        # Using copy.copy
+        msg = "shallow copies of TensorBlock are not possible, use a deepcopy instead"
+        with pytest.raises(ValueError, match=msg):
+            copy.copy(other_clone)
```

Comment:
Can't we split this in an extra test. Feels like independant. Sorry I am a fan of small tests with single asserts.

Reply:
fair point

Reply:
done!